### DISCUSSION TITLE
Welcome to FullyNoded Discussions!

### DISCUSSION THREAD

**Discussion by Fonta1n3** [Fonta1n3/FullyNoded#130] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by RobN007** [comment]
Hi,

first of all thanks for the APP! i think this is the one app i was looking for.
I managed to link my full node with the Fully Noded IOS App- i know wanted to import my hard wallet public key (xpub) but it says xpub is not supported?

Is it not possible to create a watch only wallet ? 

Thanks for your help!

**Comment by Fonta1n3** [comment]
Hi, you need to import a descriptor which holds the xpub, you can paste it in the wallet creator view. If you need help creating the descriptor let me know.

**Reply by RobN007** [reply]
Hi,

thanks for the fast reply!
Yes is there maybe a Guide how to do this ?  Is there any risk doing this ?

Thanks in advance

**Comment by Fonta1n3** [comment]
No risk, its just adding the derivation path and master key fingerprint of
your xpub so that offline signers (like Fully Noded) can sign the psbt your
watch only wallet would create.

For a bech32 (BIP84) single sig xpub the descriptor would be:

wpkh([yourMasterKeyFingerprintHere/84h/0h/0h]yourXpubHere/0/*)

If you do not know your master key fingerprint you can just put 00000000 in
its place.

If your wallet is not BIP84 then the above descriptor would not work...

**Reply by RobN007** [reply]
Thanks!

Last question - where in the app am i able to paste this created Descriptor?
Im navigating to the "Active Wallet" Overview and i can press on the "+" symbol on the top left.
I then have the options "Single-sig" "Multi-sig" and "Join Market"  - where am i able to just paste the descriptor to use it as watch only ?

**Reply by Fonta1n3** [reply]
There's three other buttons in the top right for pasting text, or QR codes. Tap the clipboard button to paste text.

**Reply by RobN007** [reply]
Im so sorry i need to ask for help again.
I found the place to paste it, but im getting an error: wpkh(): key'mypublikkey' is not valid - of course instead of mypublickey im using the xpub that is shown in my BitBox (Hard Wallet) GUI - there i can also see the "Root fingerprint" im also using this in the descriptor (i tried 00000000 also without success) and i can see that it is a bech32, p2wpkh script type and the Keypath is: m/84'/0'/0'

This is how my Descriptor is looking i tried to paste (exept of the fingerprint and xpub i am not shwoing here):

wpkh([00000000/84h/0h/0h]zpub.....xyz/0/*)

**Reply by Fonta1n3** [reply]
That's a zpub not an xpub. Is there anywhere that gives you the bip84 xpub?

**Reply by RobN007** [reply]
Hi,
unfortunately, i still haven't reached my destination  😆 

After scanning the Blockchain for the descriptor i pasted in it shows a balance of 0 - so probably still a mistake with my descriptor.
I checked the GUI of my hard wallet again to see if i got the correct xpub and here are the information i get when checking for the public adress:

I can see information for the public key for:
- Native segwit (bech32, P2WPKH)    
- Taproot (bech32m, P2TR)
- Wrapped Segwit (P2WPKH-P2SH)


Native Segwit: 
- Keypath: m/84'/0'/0'
- Public Key: starts with "zpub"

Taproot:
- Keypath:  m/86'/0'/0'
- Public Key: starts with "xpub"

Wrapped Segwit:
- Keypath:  m/49'/0'/0'
- Pubic Key: starts with "ypub"#


what i tried:

wpkh([myFingerprint/86h/0h/0h]TaprootXpubKey./0/*)   -- results in balance 0

wpkh([myFingerprint/84h/0h/0h]TaprootXpubKey./0/*)   -- results in balance 0

trying the native Segwit Public Key results in a error, probably because its zpub publicKey



I am very sorry to take up your time here but a solution that I can use Fully Noded for just checking the balance of my wallet would make me very happy! Thanks for the great support!

**Reply by Fonta1n3** [reply]
You need to convert the zpub to an xpub, you can use this tool. I thought FN would do it automatically, need to check the code, in the meantime you can use this:

https://jlopp.github.io/xpub-converter/

**Reply by RobN007** [reply]
Thanks i will check this out and test it.

The converter is giving me a different Fingerprint  - i will try using the generated xpub with the 00000000 fingerprint or should i use a different one?

**Reply by Fonta1n3** [reply]
Use what it gives you.

**Reply by RobN007** [reply]
It worked! 
As it is a watch only wallet , there is no way to use the app to make any transactions, correct?

Thank you so much for your help - great service and great app! definitely recommend
